crawler/jobs/cherrypicker_test_data.py

```diff
@@ -0,0 +1,154 @@
+)
+from crawler.helpers.general_helpers import get_config
+from crawler.main import run as run_crawler
+from crawler.types import Config
+
+
+logger = logging.getLogger(__name__)
+
+
+class TestDataError(Exception):
+    def __init__(self, message):
+        self.message = message
+
+
+def process(run_id: str, config: Config = None) -> List[List[str]]:
+    """Generates cherrypicker test data for processing by Crawler and then
+    processes it via the usual runner.
+
+    The specification of the plates to be generated should be in Mongo. Each
+    plate will contain an exact number of positive results between 0 and 96 as
+    specified. Up to 100 plates can be generated at a time.
+
+    Arguments:
+        run_id: str - The ID of the run.  If this is not found in Mongo an
+            exception will be thrown.
+
+    Returns:
+        Metadata about the plates generated, as:
+        [ [ "barcode1", "description1" ], [ "barcode2", "description2" ] ]
+    """
+    logger.info("Begin generating data.")
+
+    if config is None:
+        config, _ = get_config()
+
+    with create_mongo_client(config) as mongo_client:
+        mongo_db = get_mongo_db(config, mongo_client)
+        collection = get_mongo_collection(mongo_db, COLLECTION_CHERRYPICK_TEST_DATA)
+
+        run_doc = get_run_doc(collection, run_id)
+        if run_doc[FIELD_STATUS] != FIELD_STATUS_PENDING:
+            raise TestDataError(f"{TEST_DATA_ERROR_WRONG_STATE} '{FIELD_STATUS_PENDING}'")
+
+        try:
+            try:
+                plate_specs_string = run_doc[FIELD_PLATE_SPECS]
+                plate_specs: List[List[int]] = json.loads(plate_specs_string)
+            except (TypeError, json.JSONDecodeError):
+                raise TestDataError(TEST_DATA_ERROR_INVALID_PLATE_SPECS)
+
+            num_plates = reduce(lambda a, b: a + b[0], plate_specs, 0)
+            if num_plates < 1 or num_plates > 100:
+                raise TestDataError(TEST_DATA_ERROR_NUMBER_OF_PLATES)
+
+            positives_per_plate = [spec[1] for spec in plate_specs]
+            if any([positives < 0 or positives > 96 for positives in positives_per_plate]):
+                raise TestDataError(TEST_DATA_ERROR_NUMBER_OF_POS_SAMPLES)
+
+            update_status(collection, run_id, FIELD_STATUS_STARTED)
+
+            dt = datetime.utcnow()
+            barcodes = create_barcodes(num_plates)
+
+            update_status(collection, run_id, FIELD_STATUS_PREPARING_DATA)
+
+            csv_rows = create_csv_rows(plate_specs, dt, barcodes)
+            plates_path = os.path.join(config.DIR_DOWNLOADED_DATA, TEST_DATA_CENTRE_PREFIX)
+            plates_filename = f"{TEST_DATA_CENTRE_PREFIX}_{dt.strftime('%y%m%d_%H%M%S_%f')}.csv"
+            write_plates_file(csv_rows, plates_path, plates_filename)
+
+            update_status(collection, run_id, FIELD_STATUS_CRAWLING_DATA)
+
+            run_crawler(sftp=False, keep_files=False, add_to_dart=False, centre_prefix=TEST_DATA_CENTRE_PREFIX)
```

Comment:
add to dart decision flag needs to be passed through from UI, it is needed to be True for Beckman test data setup and false for Biosero. User decides. Or maybe we just always set it True, can discuss.